File: JS Back-End/04. NoSQL and MongoDB/MongoDB-demo/models/Person.js
const mongoose = require('mongoose');

const personSchema = new mongoose.Schema({
    firstName: { type: String, required: true },
    lastName: { type: String, required: true },
    age: {
        type: Number,
        required: true,
        min: [0, 'Age cannot be negative!']
    }
});

personSchema.methods.identify = function() {
    return `${this.firstName} ${this.lastName}`;
};

personSchema.virtual('fullName').get(function() {
    return `${this.firstName} ${this.lastName}`;
});

personSchema.methods.sayHi = function() {
    return `My name is ${this.fullName} and I am ${this.age} years old`;
};

module.exports = mongoose.model('Person', personSchema);

File: C# OOP/01. Inheritance/Workshop/Person/Person.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Person
{
    public class Person
    {
        private string name;
        protected int age;

        public Person(string name, int age)
        {
            Age = age;
            Name = name;
        }

        public int Age
        {
            get => age;
            set
            {
                if (age < 0)
                {
                    throw new Exception("age cannot be negative");
                }
                else
                {
                    age = value;
                }
            }
        }
        public string Name { get => name; set => name = value; }

        public override string ToString()
        {
            return $"Name: {Name}, Age: {Age}";
        }
    }
}
